packages/block-library/src/button/edit.native.js

```diff
@@ -0,0 +1,265 @@
+		const outlineBorderRadius = borderRadiusValue > 0 ? borderRadiusValue + styles.button.paddingTop + styles.button.borderWidth : 0;
+
+		// To achieve proper expanding and shrinking `RichText` on iOS, there is a need to set a `minWidth`
+		// value at least on 1 when `RichText` is focused or when is not focused, but `RichText` value is
+		// different than empty string.
+		const minWidth = isFocused || ( ! isFocused && text && text !== '' ) ? 1 : styles.button.minWidth;
+		// To achieve proper expanding and shrinking `RichText` on Android, there is a need to set
+		// a `placeholder` as an empty string when `RichText` is focused,
+		// because `AztecView` is calculating a `minWidth` based on placeholder text.
+		const placeholderText = isFocused ? '' : ( placeholder || __( 'Add text…' ) );
+
+		return (
+			<View
+				style={ { flex: 1 } }
+				onLayout={ this.onLayout }
+			>
+				<View
+					style={ [
+						styles.container,
+						isSelected && {
+							borderColor: this.onChangeBackgroundColor(),
+							borderRadius: outlineBorderRadius,
+							borderWidth: styles.button.borderWidth,
+						},
+					] }
+				>
+					<RichTextWrapper
+						borderRadiusValue={ borderRadiusValue }
+						backgroundColor={ this.onChangeBackgroundColor() }
+					>
+						<RichText
+							setRef={ ( richText ) => {
+								this.richTextRef = richText;
+							} }
+							placeholder={ placeholderText }
+							value={ text }
+							onChange={ ( value ) => setAttributes( { text: value } ) }
+							style={ {
+								...richTextStyle.richText,
+								color: textColor.color || '#fff',
+							} }
+							textAlign="center"
+							placeholderTextColor={ 'lightgray' }
+							identifier="content"
+							tagName="p"
+							minWidth={ minWidth }
+							maxWidth={ maxWidth }
+							id={ clientId }
+						/>
+					</RichTextWrapper>
+
+					<InspectorControls>
+						<PanelBody title={ __( 'Border Settings' ) } >
+							<RangeControl
+								label={ __( 'Border Radius' ) }
+								minimumValue={ MIN_BORDER_RADIUS_VALUE }
+								maximumValue={ MAX_BORDER_RADIUS_VALUE }
+								value={ borderRadiusValue }
+								onChange={ this.setBorderRadius }
+								separatorType="none"
+							/>
+						</PanelBody>
+						<PanelBody title={ __( 'Link Settings' ) } >
+							<ToggleControl
+								label={ __( 'Open in new tab' ) }
+								checked={ linkTarget === '_blank' }
+								onChange={ this.onToggleOpenInNewTab }
+								separatorType="fullWidth"
+							/>
+							<TextControl
+								label={ __( 'Link Rel' ) }
+								value={ url || '' }
+								valuePlaceholder={ __( 'None' ) }
+								onChange={ ( value ) => setAttributes( { url: value } ) }
+								autoCapitalize="none"
+								autoCorrect={ false }
+								separatorType="none"
+								keyboardType="url"
+							/>
+						</PanelBody>
+						<PanelBody title={ __( 'Color Settings' ) } >
+							<MissingControl
+								label={ __( 'Coming Soon' ) }
```

Comment:
This `MissingControl` seems generic enough that it would look the same regardless of where it's used. I think at least the label should be set by the control itself, and ideally the logic to open the notification as well

Reply:
Could you please elaborate more on that?

Reply:
I imagine the UI of this control would be the same here than in a different setting on another block. We could ensure it's consistent by using it as `<MissingControl />` (we might still need to specify the separator type though) and have the UI and behavior contained within the component.

Reply:
I've replaced `<MissingControl>` in favor of `<UnsupportedFooterControl>` which is displaying text passed as a label prop. In that case label is 
```
label={ __( 'Note: Theme colors are not available at this time.' ) }
```

Let me know wdyt about it ✌️